subsys/shell/shell_cmds.c

```diff
@@ -177,127 +177,137 @@ static int terminal_size_get(const struct shell *shell)
 	}
 
 	cursor_restore(shell);
-
 	return ret_val;
 }
 
-static void cmd_clear(const struct shell *shell, size_t argc, char **argv)
+static int cmd_clear(const struct shell *shell, size_t argc, char **argv)
 {
 	(void)argv;
 
 	if ((argc == 2) && (shell_help_requested(shell))) {
 		shell_help_print(shell, NULL, 0);
-		return;
+		return 0;
 	}
 	SHELL_VT100_CMD(shell, SHELL_VT100_CURSORHOME);
 	SHELL_VT100_CMD(shell, SHELL_VT100_CLEARSCREEN);
+	return 0;
 }
 
-static void cmd_shell(const struct shell *shell, size_t argc, char **argv)
+static int cmd_shell(const struct shell *shell, size_t argc, char **argv)
 {
 	(void)argv;
 
 	if ((argc == 1) || ((argc == 2) && shell_help_requested(shell))) {
 		shell_help_print(shell, NULL, 0);
-		return;
+		return 0;
 	}
 
 	shell_fprintf(shell, SHELL_ERROR, SHELL_MSG_SPECIFY_SUBCOMMAND);
+	return 0;
 }
 
-static void cmd_bacskpace_mode(const struct shell *shell, size_t argc,
+static int cmd_bacskpace_mode(const struct shell *shell, size_t argc,
 			       char **argv)
 {
 	(void)shell_cmd_precheck(shell, (argc == 2), NULL, 0);
+	return 0;
 }
 
-static void cmd_bacskpace_mode_backspace(const struct shell *shell, size_t argc,
+static int cmd_bacskpace_mode_backspace(const struct shell *shell, size_t argc,
 					 char **argv)
 {
 	if (!shell_cmd_precheck(shell, (argc == 1), NULL, 0)) {
```

Comment:
this is not that simple: precheck may also fail if help was requested. In that case return 0 is ok but if it fails because of number of arguments then it should return error.

Reply:
Yes, this is good point. It will be fixed in other PR.